sdk/python/kfp/components/_python_op.py

```diff
@@ -45,39 +46,13 @@ def _python_function_name_to_component_name(name):
     return re.sub(' +', ' ', name.replace('_', ' ')).strip(' ').capitalize()
 
 
-def _func_to_component_spec(func, extra_code='', base_image=_default_base_image) -> ComponentSpec:
-    '''Takes a self-contained python function and converts it to component
-
-    Args:
-        func: Required. The function to be converted
-        base_image: Optional. Docker image to be used as a base image for the python component. Must have python 3.5+ installed. Default is tensorflow/tensorflow:1.11.0-py3
-                    Note: The image can also be specified by decorating the function with the @python_component decorator. If different base images are explicitly specified in both places, an error is raised.
-        extra_code: Optional. Python source code that gets placed before the function code. Can be used as workaround to define types used in function signature.
-    '''
-    decorator_base_image = getattr(func, '_component_base_image', None)
-    if decorator_base_image is not None:
-        if base_image is not _default_base_image and decorator_base_image != base_image:
-            raise ValueError('base_image ({}) conflicts with the decorator-specified base image metadata ({})'.format(base_image, decorator_base_image))
-        else:
-            base_image = decorator_base_image
-    else:
-        if base_image is None:
-            raise ValueError('base_image cannot be None')
-
-    import inspect
-    import re
-    from collections import OrderedDict
-    
+def _create_component_interface_spec_from_function_signature(func) -> ComponentSpec:
```

Comment:
Why not _create_component_spec_from_function_signature?

Reply:
>How about just _extract_component()?

Sounds good. I'd like to have the `interface` word there though.
How about `_extract_component_interface`?

Reply:
Done